From cf240aed66c98c33ca4739ac8a07563325f9c3e2 Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:38:38 -0400
Subject: [virt] kvm: remove useless write to vcpu->hv_clock.tsc_timestamp

Message-id: <1406040016-3289-124-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86202
O-Subject: [RHEL7 PATCH v2 123/221] KVM: remove useless write to vcpu->hv_clock.tsc_timestamp
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

After the previous patch from Marcelo, the comment before this write
became obsolete.  In fact, the write is unnecessary.  The calls to
kvm_write_tsc ultimately result in a master clock update as soon as
all TSCs agree and the master clock is re-enabled.  This master
clock update will rewrite tsc_timestamp.

So, together with the comment, delete the dead write too.

Reviewed-by: Marcelo Tosatti <mtosatti@redhat.com>
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
(cherry picked from commit aab6d7ce37cf20753a336dc74473cf8a8aefa7c0)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/x86.c b/arch/x86/kvm/x86.c
index fc88300..0e38bd5 100644
--- a/arch/x86/kvm/x86.c
+++ b/arch/x86/kvm/x86.c
@@ -1281,8 +1281,6 @@ void kvm_write_tsc(struct kvm_vcpu *vcpu, struct msr_data *msr)
  kvm->arch.last_tsc_write = data;
  kvm->arch.last_tsc_khz = vcpu->arch.virtual_tsc_khz;
 
- /* Reset of TSC must disable overshoot protection below */
- vcpu->arch.hv_clock.tsc_timestamp = 0;
  vcpu->arch.last_guest_tsc = data;
 
  /* Keep track of which generation this VCPU has synchronized to */
-- 
1.7.1